src/main/java/org/jitsi/videobridge/cc/SimulcastController.java

```diff
@@ -1090,14 +1157,38 @@ void setOptimalIndex(int newOptimalIdx)
          */
         RawPacket[] rtpTransform(RawPacket pktIn)
         {
+            long tl0Ssrc = this.tl0Ssrc;
             if (pktIn.getSSRCAsLong() != tl0Ssrc)
```

Comment:
hmmm, what's going on here?  this used to be accessing the member variable, but now we shadow that with a local variable of the same name/value? why?

Reply:
It's so that we print the correct value in the debug logs (the field is written by the translator thread)

Reply:
you mean the call to `timeSeriesLogger.trace(...)` will be handled in another thread, where `tl0Ssrc` may have been updated before it reads it and logs?

Reply:
right, there are several threads accessing the simulcast controller, the translator thread (in which the accept method runs) and the transform threads which are driving the packets in/out of the simulcast controller. The tl0Ssrc is written by the translator thread, whereas it's read by on of the transformer threads

Reply:
right, but just so i understand, this is at least somewhat synchronized, no?  are you saying that--ignoring the trace call--`tl0ssrc` could change within the context of execution of this one method?

i believe the answer to that is no, but that the logging (where the `tl0ssrc` reference is passed) is run on yet _another_ thread, and therefore `tl0ssrc` can change before that log takes place, is that right?

Reply:
> right, but just so i understand, this is at least somewhat synchronized, no?

No, it's not synchronized (it doesn't have to be).

> are you saying that--ignoring the trace call--tl0ssrc could change within the context of execution of this one method?

Yes.

> i believe the answer to that is no, but that the logging (where the tl0ssrc reference is passed) is run on yet another thread, and therefore tl0ssrc can change before that log takes place, is that right?

`tl0Ssrc` is passed by value. The call to the logger happens in one of the packet transform threads, which only reads the field.

Reply:
Ok, so is it that previously `tl0Ssrc` was only read once, so even though it wasn't synchronized consistency wasn't a problem, but now we read it twice...once in the initial check, and then later in the log message, and now there's a chance it changes between those 2 lines so we want to save it off to prevent that?

Reply:
Yes!

Reply:
We could definitely come up with better names for a few fields, but I think doing so for the tl0Ssrc alone would leave the code in an incoherent state.

Reply:
I'm not talking about the member variable, just the new local variable you introduced.

Reply:
👍